platform/xbox/functions/_PDCLIB/_PDCLIB_w32errno.c

```diff
@@ -0,0 +1,54 @@
+/* _PDCLIB_w32errno( void )
+
+   This file is part of the Public Domain C Library (PDCLib).
+   Permission is granted to use, modify, and / or redistribute at will.
+*/
+
+#ifndef REGTEST
+#include <errno.h>
+#include <wchar.h> // Watcom bug: winnt.h assumes string.h defines wchar_t
+#include <windows.h>
+
+void _PDCLIB_w32errno(void);
+
+void _PDCLIB_w32errno(void)
+{
+    // Not exhaustive
+    switch(GetLastError()) {
+        case ERROR_SUCCESS:
+            return;
+        case ERROR_FILE_NOT_FOUND:
+        case ERROR_PATH_NOT_FOUND:
+        case ERROR_INVALID_DRIVE:
+            errno = ENOENT; break;

```

Comment:
It seems that other part of the lib don't assign errno directly but rather uses _PDCLIB_errno which in fact is a function call that returns the address of errno. Thread safety maybe ?

Reply:
`errno` is the same as `*_PDCLIB_errno()`.

Reply:
I also noticed this; in my opinion, we should stick to whatever, pdclib does / prefers.

- Will make it easier to compare to upstream and will also allow for possible changes.
- We also don't know if `_PDCLIB_errno` might be made per-thread (or something) in the future. Maybe it exists for good reasons that we don't understand.

It's a nit from me objectively (I don't think anything bad will happen). Subjectively I feel strongly (I dislike diverging from upstream).

Reply:
> We also don't know if _PDCLIB_errno might be made per-thread (or something) in the future. Maybe it exists for good reasons that we don't understand.

Yes, that's the case now in upstream PDCLib, and it's the exact reason why `errno` is the same as `*_PDCLIB_errno()`, to support this kind of change.
The example port in PDCLib seems to prefer using ``*_PDCLIB_errno()` directly, so I adjusted my code.

Reply:
If I may shine some light on this... there's a section in P.J. Plaugher's book "The Standard C Library" on the merits of using "lazy evaluation" of errno when working with the FPU. Reading the FPU status flags, he claims, basically stalls the FPU pipeline, so it would be better to have the FPU carry on _until the errno value is actually requested_, instead of setting errno after each operation.

The other point was, indeed, thread safety -- I figured I might need some initialization code to run the first time a thread queried errno.

I haven't dabbled with the FPU yet, and the thread safety turned out to be much simpler (not requiring any dynamic setup), but I didn't know that at the time I implemented `<errno.h>`.

As for what is "preferred", I haven't actually layed down a "rule" for that. In some places (platform code) I am handling **PDCLib** errno **and** native errno in the same source; here, of course I have to go through `*_PDCLIB_errno_func()` and `_PDCLIB_E...` while e.g. /usr/include/errno.h is included. I could use just plain `errno` in other source files, and am a bit torn between readability and consistency. I've shelved making any "clean-up sweeps" on errno right now because I've got so many construction sites already. But if it helps you with **your* work, I could give it a quick once-over.

Reply:
This PR is consistent now. We can adapt whatever pdclib chooses when doing a rebase in the future.